### DISCUSSION TITLE
Add option to change the local domain name from the ".lan" default

### DISCUSSION THREAD

**Discussion by rlowens** [hassio-addons/addon-adguard-home#221] [Ideas]
I just set up the DHCP server inside adguard-home addon, figure it is less likely to change than my wifi/router is. But I prefer the ".local" local domain name and AdGuard Home defaults to ".lan"

It would be nice if this was easily configurable from the addon settings.

Currently, you can manually change it like this:

From the local console (hassos inside ProxMox for me, note that the SSH addon won't get you to the hypervisor-level console):
"login" (gets you to the hypervisor-level console)
"docker exec -it addon_a0d7b954_adguard /bin/bash" (gets you to the addon's console)
vi /data/adguard/AdGuardHome.yaml (opens the file in vi editor)
downarrow to line 66 for me, the line that says "  local_domain_name: lan", rightarrow over to "lan", delete key 3 times to delete "lan", a to stard adding text, "local" (or what you want), esc, :wq
s6-svc -r /var/run/s6/services/adguard (restarts the service)

**Comment by frenck** [comment]
So, I think two things:

1. I would prefer if AdGuard itself provided the interface for those things. Making it configurable on add-on level spread configuration, which is not preferable nor improving user experience.

2. Don't use .local for DNS, it is part of / reserved space for the mDNS/Zeroconf specification, not DNS.

**Reply by rlowens** [reply]
1. I agree, that would be better but they don't (yet).
2. All I know is that mDNS/Zeroconf doesn't work on my PCs to get to the web pages of ESPHome devices via their "devicename.local" link that ESPHome provides and this seems to solve that for me. Perhaps I need to figure out mDNS/Zeroconf on Windows 10 or something.

**Reply by ghost** [reply]
Hello,

I second this request 👍 

Waiting for the AdGuard upstream update to support this, could we access from the Addon to the AdGuardHome.yaml file which according to their documentation is where it is currently supported?

https://github.com/AdguardTeam/AdGuardHome/wiki/Configuration#configuration-file
(search for local_domain_name)
![image](https://user-images.githubusercontent.com/24211083/132234605-9768d3df-d3a2-4ff5-b600-f7ec32e52253.png)

Thanks a lot!
